themes/default/content/blog/top-5-things-for-azure-devs-static-websites/index.md

```diff
@@ -0,0 +1,493 @@
+
+To see the details of the deployment, click on `Build and Deploy`.
+
+![Results](./results.png)
+
+So what happened here? First, we used the Azure Static Web Apps for Visual Studio Code to authorize access to Github. Next, we created a deployment using the extension, created a Github Actions workflow, and deployed the application. We edited the code and pushed it to the repository. The push triggered the Github Action, rebuilt, and deployed the application.
+
+The Azure Static Web Apps service lets you create a complete devops workflow.
+
+### Build and deploy with code and Pulumi
+
+Can we create a complete devops workflow with code? Of course, we can! The Pulumi Azure Native provider uses the same API as the Visual Studio. In the example below, we can duplicate the process without using an extension or the Azure CLI.
+
+```typescript
+import pulumi
+import pulumi_azure_native as azure_native
+
+static_site = azure_native.web.StaticSite("staticSite",
+    branch="master",
+    build_properties=azure_native.web.StaticSiteBuildPropertiesArgs(
+        api_location="api",
+        app_artifact_location="build",
+        app_location="app",
+    ),
+    location="West US 2",
+    name="testStaticSite0",
+    repository_token="repoToken123",
+    repository_url="https://github.com/username/RepoName",
+    resource_group_name="rg",
+    sku=azure_native.web.SkuDescriptionArgs(
+        name="Basic",
+        tier="Basic",
+    ))
+```
+
+In the example above, creating a static web app requires configuring the same parameters we used in the Visual Studio Code example. The difference is that we create a Pulumi Project with our application at the root of the project directory in a folder called `app`. Once the static web app stack is created, you can push changes and trigger a deployment. One of the advantages of using infrastructure as code is that you can use the development environment and toolchain that you prefer. You are not bound to using a specific extension or development environment. We can be productive with our toolchain and create a complete deployment workflow with Github Actions with cloud engineering and Azure Static Web Apps.
+
+## Summary
+
+This article demonstrates two ways to build static websites with Azure. The first way is to use Azure storage to deploy the website. We showed how to use the Azure Portal and infrastructure as code to deploy. The other way to deploy a static website is to use the Azure Static Web Apps Service. We used the Visual Studio Code extension to configure Github Actions to deploy the React application in this example. We demonstrated how to deploy using code because the Pulumi Azure Native provider uses the Azure API. Infrastructure as code lets us choose our development environment where we are productive. In the next installment of this series, we’ll tackle deploying Kubernetes. In the next installment of this series, we'll tackle deploying Kubernetes.
```

Comment:
```suggestion
This article demonstrates two ways to build static websites with Azure. The first way is to use Azure Storage to deploy the website. We showed how to use the Azure Portal and infrastructure as code to deploy. The other way to deploy a static website is to use the Azure Static Web Apps Service. We used the Visual Studio Code extension to configure GitHub Actions to deploy the React application in this example. We demonstrated how to deploy using code because the Pulumi Azure Native provider uses the Azure API. Infrastructure as code lets us choose our development environment where we are productive. In the next installment of this series, we’ll tackle deploying Kubernetes.
```